Generated queries file missing import for table model

### SQLDelight Version

2.0.0

### Operating System

MacOS

### Gradle Version

8.2.1

### Kotlin Version

1.9.0

### Dialect

postgresql

### AGP Version

_No response_

### Describe the Bug

When setting up a project with `deriveSchemaFromMigrations.set(true)` and putting the queries `.sq` in a different directory than the migrations `.sqm`, the generated queries file fails to compile because it doesn't have the necessary import for the table model.

This is the schema:
**src/main/sqldelight/com/example/sqldelightbug/database/V1__Initial.sqm**
```sql
CREATE TABLE Person(
  id UUID PRIMARY KEY,
  createdAt TIMESTAMPTZ NOT NULL,
  fullName TEXT NOT NULL
);
```

Queries
**src/main/sqldelight/com/example/sqldelightbug/database/queries/Person.sq**
```sql
add:
INSERT INTO Person(id, createdAt, fullName)
VALUES (?, ?, ?)
RETURNING *;
```

This is the generated data class for Person
**build/generated/sqldelight/code/Database/main/com/example/sqldelightbug/database/Person.kt**
```kotlin
package com.example.sqldelightbug.database

import java.time.OffsetDateTime
import java.util.UUID
import kotlin.String

public data class Person(
  public val id: UUID,
  public val createdAt: OffsetDateTime,
  public val fullName: String,
)
```

This Queries file fails to build because it has no import for `Person`
**build/generated/sqldelight/code/Database/main/com/example/sqldelightbug/database/queries/PersonQueries.kt**
```kotlin
package com.example.sqldelightbug.database.queries

import app.cash.sqldelight.ExecutableQuery
import app.cash.sqldelight.TransacterImpl
import app.cash.sqldelight.db.QueryResult
import app.cash.sqldelight.db.SqlCursor
import app.cash.sqldelight.db.SqlDriver
import app.cash.sqldelight.driver.jdbc.JdbcCursor
import app.cash.sqldelight.driver.jdbc.JdbcPreparedStatement
import java.time.OffsetDateTime
import java.util.UUID
import kotlin.Any
import kotlin.String

public class PersonQueries(
  driver: SqlDriver,
) : TransacterImpl(driver) {
  public fun <T : Any> add(
    id: UUID,
    createdAt: OffsetDateTime,
    fullName: String,
    mapper: (
      id: UUID,
      createdAt: OffsetDateTime,
      fullName: String,
    ) -> T,
  ): ExecutableQuery<T> = AddQuery(id, createdAt, fullName) { cursor ->
    check(cursor is JdbcCursor)
    mapper(
      cursor.getObject<UUID>(0)!!,
      cursor.getObject<OffsetDateTime>(1)!!,
      cursor.getString(2)!!
    )
  }

  public fun add(
    id: UUID,
    createdAt: OffsetDateTime,
    fullName: String,
  ): ExecutableQuery<Person> = add(id, createdAt, fullName) { id_, createdAt_, fullName_ ->
    Person(
      id_,
      createdAt_,
      fullName_
    )
  }

  private inner class AddQuery<out T : Any>(
    public val id: UUID,
    public val createdAt: OffsetDateTime,
    public val fullName: String,
    mapper: (SqlCursor) -> T,
  ) : ExecutableQuery<T>(mapper) {
    override fun <R> execute(mapper: (SqlCursor) -> QueryResult<R>): QueryResult<R> =
        driver.executeQuery(-1_504_056_887, """
    |INSERT INTO Person(id, createdAt, fullName)
    |VALUES (?, ?, ?)
    |RETURNING *
    """.trimMargin(), mapper, 3) {
      check(this is JdbcPreparedStatement)
      bindObject(0, id)
      bindObject(1, createdAt)
      bindString(2, fullName)
    }

    override fun toString(): String = "Person.sq:add"
  }
}
```

### Stacktrace

```shell
./gradlew build

> Task :compileKotlin FAILED
e: file:///Users/gustav.karlsson/Development/sqldelightbug/build/generated/sqldelight/code/Database/main/com/example/sqldelightbug/database/queries/PersonQueries.kt:40:22 Unresolved reference: Person
e: file:///Users/gustav.karlsson/Development/sqldelightbug/build/generated/sqldelight/code/Database/main/com/example/sqldelightbug/database/queries/PersonQueries.kt:41:5 Unresolved reference: Person

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.compilerRunner.GradleCompilerRunnerWithWorkers$GradleKotlinCompilerWorkAction
   > Compilation error. See log for more details

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 729ms
2 actionable tasks: 1 executed, 1 up-to-date
```


### Gradle Build Script

```gradle
import org.jetbrains.kotlin.gradle.tasks.KotlinCompile

plugins {
    kotlin("jvm") version "1.9.0"
    id("app.cash.sqldelight") version "2.0.0"
}

repositories {
    mavenCentral()
}

dependencies {
    implementation("org.postgresql:postgresql:42.6.0")
    implementation("app.cash.sqldelight:jdbc-driver:2.0.0")
}

tasks.test {
    useJUnitPlatform()
}

java {
    targetCompatibility = JavaVersion.VERSION_1_8
}

tasks.withType<KotlinCompile> {
    kotlinOptions.jvmTarget = JavaVersion.VERSION_1_8.toString()
}

sqldelight {
    databases {
        // Name of the generated database class
        create("Database") {
            // Where the generated Database class is created
            packageName.set("com.example.sqldelightbug.database")
            dialect("app.cash.sqldelight:postgresql-dialect:2.0.0")
            deriveSchemaFromMigrations.set(true)
        }
    }
}
```

--- a/sqldelight-compiler/dialect/src/main/kotlin/app/cash/sqldelight/dialect/api/ReturningQueryable.kt
+++ b/sqldelight-compiler/dialect/src/main/kotlin/app/cash/sqldelight/dialect/api/ReturningQueryable.kt
@@ -29,7 +29,7 @@ class ReturningQueryable(
       ?: return@lazy null
     val requestedColumnsAreIdenticalToTable = table.query.columns.flattenCompounded() == pureColumns
     if (requestedColumnsAreIdenticalToTable) {
-      tableName
+      table.tableName
     } else {
       null
     }

--- a/sqldelight-compiler/src/test/kotlin/app/cash/sqldelight/core/migrations/MigrationQueryTest.kt
+++ b/sqldelight-compiler/src/test/kotlin/app/cash/sqldelight/core/migrations/MigrationQueryTest.kt
@@ -25,6 +25,10 @@ class MigrationQueryTest {
     checkFixtureCompiles("alter-table-alter-column", PostgreSqlDialect())
   }
 
+  @Test fun `varying query migration packages`() {
+    checkFixtureCompiles("varying-query-migration-packages", PostgreSqlDialect())
+  }
+
   private fun checkFixtureCompiles(fixtureRoot: String, dialect: SqlDelightDialect = SqliteDialect()) {
     val result = FixtureCompiler.compileFixture(
       overrideDialect = dialect,

--- a/sqldelight-compiler/src/test/migration-interface-fixtures/varying-query-migration-packages/com/example/migrations/1.sqm
+++ b/sqldelight-compiler/src/test/migration-interface-fixtures/varying-query-migration-packages/com/example/migrations/1.sqm
@@ -1,0 +1,4 @@
+CREATE TABLE test (
+  first TEXT NOT NULL,
+  second INT NOT NULL
+);

--- a/sqldelight-compiler/src/test/migration-interface-fixtures/varying-query-migration-packages/com/example/queries/Data.sq
+++ b/sqldelight-compiler/src/test/migration-interface-fixtures/varying-query-migration-packages/com/example/queries/Data.sq
@@ -1,0 +1,12 @@
+migrationSelect:
+SELECT *
+FROM test;
+
+migrationInsert:
+INSERT INTO test(first, second) VALUES (?, ?) RETURNING *;
+
+migrationDelete:
+DELETE FROM test WHERE first = ? RETURNING *;
+
+migrationUpdate:
+UPDATE test SET first = ? RETURNING *;

--- a/sqldelight-compiler/src/test/migration-interface-fixtures/varying-query-migration-packages/output/com/example/migrations/Test.kt
+++ b/sqldelight-compiler/src/test/migration-interface-fixtures/varying-query-migration-packages/output/com/example/migrations/Test.kt
@@ -1,0 +1,9 @@
+package com.example.migrations
+
+import kotlin.Int
+import kotlin.String
+
+public data class Test(
+  public val first: String,
+  public val second: Int,
+)

--- a/sqldelight-compiler/src/test/migration-interface-fixtures/varying-query-migration-packages/output/com/example/queries/DataQueries.kt
+++ b/sqldelight-compiler/src/test/migration-interface-fixtures/varying-query-migration-packages/output/com/example/queries/DataQueries.kt
@@ -1,0 +1,135 @@
+package com.example.queries
+
+import app.cash.sqldelight.ExecutableQuery
+import app.cash.sqldelight.Query
+import app.cash.sqldelight.TransacterImpl
+import app.cash.sqldelight.db.QueryResult
+import app.cash.sqldelight.db.SqlCursor
+import app.cash.sqldelight.db.SqlDriver
+import app.cash.sqldelight.driver.jdbc.JdbcCursor
+import app.cash.sqldelight.driver.jdbc.JdbcPreparedStatement
+import com.example.migrations.Test
+import kotlin.Any
+import kotlin.Int
+import kotlin.String
+
+public class DataQueries(
+  driver: SqlDriver,
+) : TransacterImpl(driver) {
+  public fun <T : Any> migrationSelect(mapper: (first: String, second: Int) -> T): Query<T> =
+      Query(-1_568_224_787, arrayOf("test"), driver, "Data.sq", "migrationSelect", """
+  |SELECT *
+  |FROM test
+  """.trimMargin()) { cursor ->
+    check(cursor is JdbcCursor)
+    mapper(
+      cursor.getString(0)!!,
+      cursor.getInt(1)!!
+    )
+  }
+
+  public fun migrationSelect(): Query<Test> = migrationSelect { first, second ->
+    Test(
+      first,
+      second
+    )
+  }
+
+  public fun <T : Any> migrationInsert(
+    first: String,
+    second: Int,
+    mapper: (first: String, second: Int) -> T,
+  ): ExecutableQuery<T> = MigrationInsertQuery(first, second) { cursor ->
+    check(cursor is JdbcCursor)
+    mapper(
+      cursor.getString(0)!!,
+      cursor.getInt(1)!!
+    )
+  }
+
+  public fun migrationInsert(first: String, second: Int): ExecutableQuery<Test> =
+      migrationInsert(first, second) { first_, second_ ->
+    Test(
+      first_,
+      second_
+    )
+  }
+
+  public fun <T : Any> migrationDelete(first: String, mapper: (first: String, second: Int) -> T):
+      ExecutableQuery<T> = MigrationDeleteQuery(first) { cursor ->
+    check(cursor is JdbcCursor)
+    mapper(
+      cursor.getString(0)!!,
+      cursor.getInt(1)!!
+    )
+  }
+
+  public fun migrationDelete(first: String): ExecutableQuery<Test> = migrationDelete(first) {
+      first_, second ->
+    Test(
+      first_,
+      second
+    )
+  }
+
+  public fun <T : Any> migrationUpdate(first: String, mapper: (first: String, second: Int) -> T):
+      ExecutableQuery<T> = MigrationUpdateQuery(first) { cursor ->
+    check(cursor is JdbcCursor)
+    mapper(
+      cursor.getString(0)!!,
+      cursor.getInt(1)!!
+    )
+  }
+
+  public fun migrationUpdate(first: String): ExecutableQuery<Test> = migrationUpdate(first) {
+      first_, second ->
+    Test(
+      first_,
+      second
+    )
+  }
+
+  private inner class MigrationInsertQuery<out T : Any>(
+    public val first: String,
+    public val second: Int,
+    mapper: (SqlCursor) -> T,
+  ) : ExecutableQuery<T>(mapper) {
+    override fun <R> execute(mapper: (SqlCursor) -> QueryResult<R>): QueryResult<R> =
+        driver.executeQuery(-1_845_995_606,
+        """INSERT INTO test(first, second) VALUES (?, ?) RETURNING *""", mapper, 2) {
+      check(this is JdbcPreparedStatement)
+      bindString(0, first)
+      bindInt(1, second)
+    }
+
+    override fun toString(): String = "Data.sq:migrationInsert"
+  }
+
+  private inner class MigrationDeleteQuery<out T : Any>(
+    public val first: String,
+    mapper: (SqlCursor) -> T,
+  ) : ExecutableQuery<T>(mapper) {
+    override fun <R> execute(mapper: (SqlCursor) -> QueryResult<R>): QueryResult<R> =
+        driver.executeQuery(-1_997_661_540, """DELETE FROM test WHERE first = ? RETURNING *""",
+        mapper, 1) {
+      check(this is JdbcPreparedStatement)
+      bindString(0, first)
+    }
+
+    override fun toString(): String = "Data.sq:migrationDelete"
+  }
+
+  private inner class MigrationUpdateQuery<out T : Any>(
+    public val first: String,
+    mapper: (SqlCursor) -> T,
+  ) : ExecutableQuery<T>(mapper) {
+    override fun <R> execute(mapper: (SqlCursor) -> QueryResult<R>): QueryResult<R> =
+        driver.executeQuery(-1_501_049_414, """UPDATE test SET first = ? RETURNING *""", mapper, 1)
+        {
+      check(this is JdbcPreparedStatement)
+      bindString(0, first)
+    }
+
+    override fun toString(): String = "Data.sq:migrationUpdate"
+  }
+}

